DisguiseUnityRenderStream/Runtime/PipelineAbstraction/DepthCopy.cs

```diff
@@ -29,139 +28,62 @@ public enum Mode
-                }
-            }
-
-            public void Create(string shaderPath, string shaderPass)
-            {
-                m_shader = Shader.Find(shaderPath);
-                if (m_shader != null)
-                {
-                    m_material = CoreUtils.CreateEngineMaterial(m_shader);
-                    m_pass = m_material.FindPass(shaderPass);
-                }
-            }
+        const string k_profilerTag = "Disguise Depth Copy";
 
-            public void Dispose()
-            {
-                CoreUtils.Destroy(m_material);
-            }
-        }
+        public static DepthCopy instance { get; }
 
-        public Mode mode
+        static DepthCopy()
         {
-            get => m_mode;
-            set
-            {
-                m_mode = value;
-                
-                m_shaderResources = m_mode switch {
-                    Mode.Raw => s_shaderVariantResources.m_Raw,
-                    Mode.Eye => s_shaderVariantResources.m_Eye,
-                    Mode.Linear01 => s_shaderVariantResources.m_linear01,
-                    _ => throw new ArgumentOutOfRangeException()
-                };
-            }
+            instance = new DepthCopy();
         }
         
-        const string k_profilerTag = "Disguise Depth Copy";
-        const string k_ShaderPass = "Depth Copy";
-
-        static readonly ShaderVariantResources s_shaderVariantResources;
+        readonly Material m_material;
 
-        Mode m_mode = Mode.Linear01;
-        ShaderResources m_shaderResources;
-        
-        static DepthCopy()
+        DepthCopy()
         {
-#if DISGUISE_CAPTURE_DEPTH
 #if !(UNITY_PIPELINE_HDRP && HDRP_VERSION_SUPPORTED) && !(UNITY_PIPELINE_URP && URP_VERSION_SUPPORTED)
             Debug.LogError($"No supported render pipeline was found for {nameof(DepthCopy)}.");
 #endif
             
-            s_shaderVariantResources.Create(k_ShaderName, k_ShaderPass);
-            Assert.IsTrue(s_shaderVariantResources.IsLoaded, $"Couldn't load the shader resources for {nameof(DepthCopy)}");
-#endif
-        }
-
-        public DepthCopy()
-        {
-#if DISGUISE_CAPTURE_DEPTH
-            mode = Mode.Linear01;
-#endif
+            var shader = Shader.Find(k_ShaderName);
+            if (shader != null)
+            {
+                m_material = CoreUtils.CreateEngineMaterial(shader);
+            }
+            
+            Assert.IsTrue(shader != null && m_material != null, $"Couldn't load the shader resources for {nameof(DepthCopy)}");
         }
 
         /// <summary>
         /// Performs the copy using the SRP's currently active camera and the provided <see cref="FrameData"/>.
         /// </summary>
         /// <param name="depthOutput">The texture to which the depth will be written to.</param>
-        public void Execute(ScriptableRenderContext context, RenderTexture depthOutput)
+        public void Execute(ScriptableRenderContext context, RenderTexture depthOutput, Mode mode)
```

Comment:
Minor: context and mode don't have their entry in the method's description but depthOutput has.